### ISSUE TITLE
Remove need to specify file name when copying to a directory

### ISSUE THREAD

**Issue by HirschBerge** [thewh1teagle/mc#11]
**Expected behavior**
Ability to copy a file to a directory (empty or not) without having to type out the /filename.ext again. Anything with "modern copying" in the name should intuit this
```bash
target/release/mc file.iso /mnt/storage/Files
Copy file.iso to /mnt/storage/Files/file.iso...
  [00:00:31] [###########] 20.00 GiB/20.00 GiB (00:00:00)
Copy completed successfully in 31.452593323s. <-- the name should probably be formatted to like 2 digits too
```
current behavior:
```bash
target/release/mc file.iso /mnt/storage/Files/
2024-12-15T14:52:04.129327Z ERROR mc: Error: Destination '/mnt/storage/Files/' already exists. Use -f to overwrite.
```

**Comment by thewh1teagle** [comment]
Fixed

**Comment by HirschBerge** [comment]
Wait, found a bug introduced by this I believe.
```
thread 'main' panicked at src/main.rs:53:85:
called `Result::unwrap()` on an `Err` value: No such file or directory (os error 2)

Location:
    src/file_ops.rs:79:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
Looks like it's coming from [here](https://github.com/thewh1teagle/mc/blob/955cf68b1b36f3913a8ad5ef592d752de95d547d/src/files.rs#L10)

Without really being super familiar with the codebase, I think the logic should be check if the file/dir exists, prompting to overwrite if it does, then check if the parent dir doesn't exists, erroring out if it doesn't.

Depending on vision, you could alternatively just recursively create the path, but I'm never sure if that's the best course

**Comment by thewh1teagle** [comment]
Can you use the latest version? see the updated readme now we have updater so it's easy to update.

**Comment by HirschBerge** [comment]
> Can you use the latest version? see the updated readme now we have updater so it's easy to update.

I don't use your binaries.

I just git pulled again, you're still using an unwrap instead of handling the Result here
```
thread 'main' panicked at src/main.rs:53:82:
```

**Comment by thewh1teagle** [comment]
> I don't use your binaries.
> 
> I just git pulled again

Even better. so reinstall it from git because it seems like you used older version.

**Comment by thewh1teagle** [comment]
```console
dd if=/dev/zero of=dummy bs=1G count=1
mc dummy copied_dummy --verify
```
This one works for you?

**Comment by HirschBerge** [comment]
> ```
> dd if=/dev/zero of=dummy bs=1G count=1
> mc dummy copied_dummy --verify
> ```
> 
> This one works for you?

this does not at all represent what I've been telling you



> > I don't use your binaries.
> > I just git pulled again
> 
> Even better. so reinstall it from git because it seems like you used older version.

You "fixed" the unwrap in [5979098](https://github.com/thewh1teagle/mc/commit/5979098ed11c1722ac818324f9698b40897fe855) which was pushed well after you originally told me you fixed it before.

You are not addressing or even commenting on my suggested logical changes. You seem to have a large void in your knowledge of what a *good* replacement of `cp` or `rsync` would need without the desire to learn. So I'm no longer interested in this project.

**Comment by thewh1teagle** [comment]
> You are not addressing or even commenting on my suggested logical changes. 

Then what I did here?


> Ability to copy a file to a directory (empty or not) without having to type out the /filename.ext again. Anything with "modern copying" in the name should intuit this

I fixed it



> You seem to have a large void in your knowledge of what a _good_ replacement of `cp` or `rsync` would need without the desire to learn.

If you really care you can contribute instead of pretend to explain well.



> So I'm no longer interested in this project.

Good for me. next time don't try to help if you don't have the patience.

Regarding the unwraps: it's CLI app. it doesn't matter if I unwrap or propagate error if it's in **main function**.

Thanks for you kind words. don't use this.
